### ISSUE TITLE
Values yielded by imap inconsistent with map

### ISSUE THREAD

**Issue by spyoungtech** [kennethreitz/grequests#111]
Currently, `imap` will not yield values returned by the exception handler and will not yield `None` values. However, responses from `map` will include values returned by the exception handler and will include `None` values.

Is there any reason why `imap` shouldn't be consistent with `map` in this way, at least for the exception handler?

Forgive me if I'm missing something, but I would have thought something like this would be appropriate.

```python
    for request in pool.imap_unordered(send, requests):
        if request.response is not None:
            yield request.response
        elif exception_handler:
            yield exception_handler(request, request.exception)
        else: # probably leave this clause out
            yield None
```

In other words, `a` and `b` should be the same in this example.

```py
import grequests

req_list = [grequests.get('http://fakedomain')]

def ex_handler(req, ex):
    return "foo"

a = grequests.map(req_list, exception_handler=ex_handler)
b = [item for item in grequests.imap(req_list, exception_handler=ex_handler)]
```

But they are not

```python
>>> a = grequests.map(req_list, exception_handler=ex_handler)
>>> b = [item for item in grequests.imap(req_list, exception_handler=ex_handler)]
>>> a
['foo']
>>> b
[]
```

**Comment by BencoLee** [comment]
Yeah, I have notice that. I don't think `exception_handler` for all responses is a good option as well.

**Comment by spyoungtech** [comment]
If no behavior change is warranted, perhaps some documentation on this behavior.  The way the readme is worded right now had given me the impression that the results for `map` and `imap` would be identical, just unordered.

There may very well be a good reason for it, but if there is, it's not immediately apparent to me. The current logic for `map` is this

```python
    for request in requests:
        if request.response is not None:
            ret.append(request.response)
        elif exception_handler and hasattr(request, 'exception'):
            ret.append(exception_handler(request, request.exception))
        else:
            ret.append(None)
```

So why not the same for imap? Its counterpart logic doesn't yield the exception results. Not clear to me why this is.

```py
    for request in pool.imap_unordered(send, requests):
        if request.response is not None:
            yield request.response
        elif exception_handler:
            exception_handler(request, request.exception) # why not yield this?
```

In my situation, I had worked out a solution to a problem using `map` (but cautiously not relying on order) where the `exception_handler` would produce a useful value. To my surprise, when switching to `imap`, it didn't produce the same values :(

**Comment by spyoungtech** [comment]
addressed in #126
